tools/geom.go

```diff
@@ -0,0 +1,450 @@
+package tools
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"math"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/USACE/filestore"
+	"github.com/dewberry/gdal"
+)
+
+// GeomFileContents keywords  and data container for ras flow file search
+type GeomFileContents struct {
+	Path           string
+	FileExt        string //`json:"File Extension"`
+	GeomTitle      string //`json:"Geom Title"`
+	ProgramVersion string //`json:"Program Version"`
+	Description    string //`json:"Description"`
+}
+
+// getGeomData Reads a geometry file. returns none to allow concurrency
+func getGeomData(rm *RasModel, fn string, wg *sync.WaitGroup, errChan chan error) {
+
+	defer wg.Done()
+
+	meta := GeomFileContents{Path: fn, FileExt: filepath.Ext(fn)}
+
+	f, err := rm.FileStore.GetObject(fn)
+	if err != nil {
+		errChan <- err
+		return
+	}
+	defer f.Close()
+
+	sc := bufio.NewScanner(f)
+	var line string
+	header := true
+	for sc.Scan() {
+
+		line = sc.Text()
+
+		match, err := regexp.MatchString("=", line)
+
+		if err != nil {
+			errChan <- err
+			return
+		}
+
+		beginDescription, err := regexp.MatchString("BEGIN GEOM DESCRIPTION", line)
+
+		if err != nil {
+			errChan <- err
+			return
+		}
+
+		if match {
+			data := strings.Split(line, "=")
+
+			switch data[0] {
+
+			case "Geom Title":
+				meta.GeomTitle = data[1]
+
+			case "Program Version":
+				meta.ProgramVersion = data[1]
+
+			case "River Reach", "Storage Area":
+				header = false
+
+			}
+
+		} else if header && beginDescription {
+
+			for sc.Scan() {
+				line = sc.Text()
+				endDescription, _ := regexp.MatchString("END GEOM DESCRIPTION", line)
+
+				if endDescription {
+					break
+
+				} else {
+					if line != "" {
+						meta.Description += line + "\n"
+					}
+				}
+
+			}
+
+		}
+	}
+
+	rm.Metadata.GeomFiles = append(rm.Metadata.GeomFiles, meta)
+	return
+}
+
+// sourceCRS
+var sourceCRS string = `PROJCS["NAD_1983_StatePlane_Texas_Central_FIPS_4203_Feet",GEOGCS["GCS_North_American_1983",DATUM["D_North_American_1983",SPHEROID["GRS_1980",6378137.0,298.257222101]],PRIMEM["Greenwich",0.0],UNIT["Degree",0.0174532925199433]],PROJECTION["Lambert_Conformal_Conic"],PARAMETER["False_Easting",2296583.333333333],PARAMETER["False_Northing",9842500.0],PARAMETER["Central_Meridian",-100.3333333333333],PARAMETER["Standard_Parallel_1",30.11666666666667],PARAMETER["Standard_Parallel_2",31.88333333333333],PARAMETER["Latitude_Of_Origin",29.66666666666667],UNIT["Foot_US",0.3048006096012192]]`
```

Comment:
What do you think about moving stuff like this and the 4326 below into config so that it is easier to find/change later?

Reply:
In another branch, which is ahead of this branch, the sourceCRS is read directly from a .projection file and passed through to the GetGeospatialData function. The destinationCRS is specified within the model.go file. However, I like the idea of adding the destinationCRS to the configuration file. I have added this as issue #4 and will address it on the other branch.